hw19 Использование Hibernate

Taisiya-Boon opened the pull request.

Taisiya-Boon committed:
hw19 Использование Hibernate

stvort commented on hw19-hibernate/src/main/java/ru/otus/DbServiceDemo.java:
```diff
@@ -0,0 +1,43 @@
+
+import java.util.Optional;
+
+public class DbServiceDemo {
```
Демо не работает

stvort commented on hw19-hibernate/src/main/java/ru/otus/core/model/AddressDataSet.java:
```diff
@@ -0,0 +1,68 @@
+
+    @OneToOne
+    @JoinColumn(name = "user_id")
+    public User user;
```
Лучше `private`

stvort commented on hw19-hibernate/src/main/java/ru/otus/core/model/PhoneDataSet.java:
```diff
@@ -0,0 +1,75 @@
+    private String number;
+
+    @ManyToOne(cascade = CascadeType.ALL)
+    public User user;
```
И тут)

stvort commented on hw19-hibernate/src/main/java/ru/otus/hibernate/dao/UserDaoHibernate.java:
```diff
@@ -0,0 +1,67 @@
+    DatabaseSessionHibernate currentSession = sessionManager.getCurrentSession();
+    try {
+      Optional<User> userOptional = Optional.ofNullable(currentSession.getHibernateSession().find(User.class, id));
+      Query q = currentSession.getHibernateSession().createQuery("From PhoneDataSet ");
```
Так мы назначим пользователю все телефоны из БД. Даже те, что к нему не относятся. По заданию чтение тоже должно происходить каскадно. Т.е. `find` должен вернуть уже юзера со всем содержимым. И он так и сделает, если пользователя корректно сохранить. Нам это мешает сделать двунаправленная связь. Это не означает, что ее надо обязательно убрать. Но учесть этот момент при сохранении нужно

Taisiya-Boon committed:
hw19  в классе User изменён конструктор
      в классах AddressDataSet, PhoneDataSet изменён модификатор поля user на private
      DbServiceDemo теперь работает

stvort commented on hw19-hibernate/src/main/java/ru/otus/core/model/User.java:
```diff
@@ -0,0 +1,99 @@
+    @Column(name = "name")
+    private String name;
+
+    @OneToOne(cascade = CascadeType.ALL)
+    private AddressDataSet addressDataSet;
```
По заданию не должно быть update-ов. При сохранении юзера в логах есть:
```
update addressDataSet set street=?, user_id=? where id=?
```

Taisiya-Boon committed:
hw19 Использование Hibernate
     при сохранении объектов в БД, не происходит update-ов

Taisiya-Boon committed:
Merge branch 'master' into hw19-hibernate
